src/app/components/article/ExplainerForm.js

```diff
@@ -0,0 +1,174 @@
+import React from 'react';
+import Relay from 'react-relay/classic';
+import { graphql, commitMutation } from 'react-relay/compat';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import ArticleForm from './ArticleForm';
+import { withSetFlashMessage } from '../FlashMessage';
+
+const createMutation = graphql`
+  mutation ExplainerFormCreateExplainerMutation($input: CreateExplainerInput!) {
+    createExplainer(input: $input) {
+      explainer {
+        id
+        dbid
+        title
+        description
+        url
+        language
+        tags
+        user {
+          dbid
+          name
+        }
+        team {
+          dbid
+          slug
+        }
+      }
+    }
+  }
+`;
+
+const updateMutation = graphql`
+  mutation ExplainerFormUpdateExplainerMutation($input: UpdateExplainerInput!) {
+    updateExplainer(input: $input) {
+      explainer {
+        id
+        dbid
+        title
+        description
+        url
+        language
+        user {
+          dbid
+          name
+        }
+        team {
+          dbid
+          slug
+        }
+      }
+    }
+  }
+`;
+
+const ExplainerForm = ({
+  article,
+  team,
+  onClose,
+  setFlashMessage,
+}) => {
+  const type = article?.dbid ? 'edit' : 'create';
+  const [saving, setSaving] = React.useState(Boolean(false));
+  const [error, setError] = React.useState(Boolean(false));
+  const [explainer, setExplainer] = React.useState({
+    title: article?.title || null,
+    description: article?.descripton || null,
+    language: article?.language || null,
+    tags: article?.tags || [],
+    url: article?.url || null,
+  });
+
+  const handleSuccess = () => {
+    setSaving(false);
+    setError(false);
+    onClose(false);
+    setFlashMessage(<FormattedMessage
+      id="explainerForm.saveSuccess"
+      defaultMessage="Article created successfully!"
+      description="Save item action success message"
+    />, 'success');
+  };
+
+  const handleError = () => {
+    setSaving(false);
+    setError(false);
+    onClose(false);
+    setFlashMessage(<FormattedMessage
+      id="explainerForm.saveError"
+      defaultMessage="Error with saving your article"
+      description="save item action failure message"
+    />, 'error');
+  };
```

Comment:
Ideally we want to display any errors coming from the backend and fallback to a standard generic error message if that's not possible